ui/src/components/admin/sermonNotes/SermonNotesEditorModal.js

```diff
@@ -0,0 +1,384 @@
+  const { user, editSermonNotesData, actionOnEditor, setIsEditorOpen } = props;
+  const {
+    register,
+    handleSubmit,
+    reset,
+    setValue,
+    formState: { errors },
+  } = useForm();
+  const toast = useToast();
+
+  // Create a object
+  const sermonIdMap = {
+    'Sunday Celebration': 'sn',
+    Encounter: 'encounter',
+  };
+
+  const [title, setTitle] = useState(undefined);
+  const [sermondId, setSermonId] = useState(undefined);
+  const [subtitle, setSubtitle] = useState(undefined);
+  const [speaker, setSpeaker] = useState(undefined);
+  const [sermonSeries, setSermonSeries] = useState(undefined);
+  const [imageLink, setImageLink] = useState(undefined);
+  const [originalContent, setOriginalContent] = useState(undefined);
+  const [sermonLink, setSermonLink] = useState(undefined);
+  const [serviceType, setServiceType] = useState(undefined);
+  const [passage, setPassage] = useState(undefined);
+  const [date, setDate] = useState(undefined);
+
+  // This is the state that will hold the number of sermons for a particular date for edge case handling
+  const [numberOfSermons, setNumberOfSermons] = useState(0);
+
+  const editorSubmitButton = (actionOnEditor) => {
+    switch (actionOnEditor) {
+      default:
+        return 'Create Sermon Notes';
+      case 'edit':
+        return 'Update Sermon Notes';
+      case 'duplicate':
+        return 'Duplicate Sermon Notes';
+    }
+  };
+
+  const setSermonNotesData = (data) => {
+    if (data) setValue('title', data.title);
+    setValue('subtitle', data.subtitle);
+    setValue('speaker', data.speaker);
+    setValue('sermonSeries', data.sermonSeries);
+    setValue('date', data.date);
+    setValue('imageLink', data.imageLink);
+    setValue('originalContent', data.originalContent);
+    setValue('sermonLink', data.sermonLink);
+    setValue('serviceType', data.serviceType);
+    setValue('passage', data.passage);
+
+    setTitle(data.title);
+    setSubtitle(data.subtitle);
+    setSpeaker(data.speaker);
+    setSermonSeries(data.sermonSeries);
+    setDate(data.date);
+    setImageLink(data.imageLink);
+    setOriginalContent(data.originalContent);
+    setSermonLink(data.sermonLink);
+    setServiceType(data.serviceType);
+    setPassage(data.passage);
+  };
+
+  const onSubmit = async (data) => {
+    setSermonNotesData(data);
+  };
+
+  const formatDate = (date) => {
+    const parts = date.split('-');
+    const day = parts[2];
+    const month = parts[1];
+    const year = parts[0];
+    return `${day}${month}${year}`;
+  };
+
+  const fetchSermonNotes = async () => {
+    try {
+      const { data } = await axios.get('/api/sermon-notes-parent/get');
+      return data;
+    } catch (err) {
+      console.log(err);
+      toast({
+        description:
+          'There was an issue with the request, please talk to a t3ch support',
+        status: 'error',
+        duration: 8000,
+        isClosable: true,
+      });
+    }
+  };
```

Comment:
if its the exact same function from the prev file, you can just export that function and import it here

Reply:
Not really the same, that has react hooks used